homeassistant/components/opencv.py

```diff
@@ -0,0 +1,229 @@
+"""
+Support for OpenCV image/video processing.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/opencv/
+"""
+import asyncio
+import logging
+import os
+import voluptuous as vol
+
+from homeassistant.helpers import (
+    discovery,
+    config_validation as cv,
+)
+
+from homeassistant.const import (
+    CONF_NAME,
+    CONF_ENTITY_ID,
+    CONF_FILE_PATH
+)
+
+REQUIREMENTS = ['opencv-python==3.2.0.6', 'numpy==1.12.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_MATCH_NAME = 'name'
+ATTR_MATCH_ID = 'id'
+ATTR_MATCH_COORDS = 'coords'
+ATTR_MATCH_REGIONS = 'regions'
+ATTR_MATCHES = 'matches'
+
+BASE_PATH = os.path.realpath(__file__)
+
+CONF_CLASSIFIER = 'classifier'
+CONF_COLOR = 'color'
+CONF_MIN_SIZE = 'min_size'
+CONF_SCALE = 'scale'
+CONF_NEIGHBORS = 'neighbors'
+
+DATA_CLASSIFIER_GROUPS = 'classifier_groups'
+
+DEFAULT_CLASSIFIER_PATH = \
+    os.path.join(os.path.dirname(BASE_PATH),
+                 'opencv_classifiers',
+                 'lbpcascade_frontalface.xml')
+DEFAULT_CLASSIFIER = [{
+    CONF_FILE_PATH: DEFAULT_CLASSIFIER_PATH,
+    CONF_NAME: 'Face',
+    CONF_MIN_SIZE: (30, 30),
+    CONF_SCALE: 1.1,
+    CONF_NEIGHBORS: 4
+}]
+DEFAULT_NAME = 'OpenCV'
+DEFAULT_MIN_SIZE = (30, 30)
+DEFAULT_SCALE = 1.1
+DEFAULT_NEIGHBORS = 4
+
+DOMAIN = 'opencv'
+
+CLASSIFIER_GROUP_CONFIG = {
+    vol.Optional(CONF_CLASSIFIER, default=DEFAULT_CLASSIFIER): vol.All(
+        cv.ensure_list,
+        [vol.Schema({
+            vol.Optional(CONF_FILE_PATH, default=DEFAULT_CLASSIFIER_PATH):
+                cv.isfile,
+            vol.Optional(CONF_NAME, default=DEFAULT_NAME):
```

Comment:
Is it obvious what the difference is between the two `name` keys in the different levels of the config tree is? Perhaps consider having some identifier added to `name` here?